tools/print_memory_usage.py

```diff
@@ -0,0 +1,386 @@
+#!/usr/bin/env python3
```

Comment:
Reminder: Copyright header.

Reply:
Yeah, just put that in. Realized it when I started reviewing your PR. Thanks.